### PR TITLE
MCP-2137 Add sufficientEvidence flag to assessment results.

### PR DESCRIPTION
**Pull request by ianwolson**
MCP-2137 Add sufficientEvidence flag to assessment results.

## What was the problem?
We need to add a sufficient evidence flag to claims after MAS automated claim processes the claims.

Associated tickets or Slack threads:

- [MCP-2137](https://amida.atlassian.net/browse/MCP-2137)

## How to test this PR
- Step 1 Run MAS automated claim endpoint
- Step 2 Allow the MAS endpoint to do its thing, it takes about a minute.
- Step 3 Check the assessment result in the VRO database for the correct flag, [] = false and [v] = true.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by au2 — CHANGES_REQUESTED** [review]
I am assuming we should follow what we did for version 1 path and move this flag to contentions. I was under the impression that database was designed for multiple contentions per claim for future multiple issue claims. Either way this is a assesment result item not claim since it is being produced by assessment,

**Comment by dimitri-amida** [comment]
@ianwolson in latest development, I created MasAssessmentResultProcessor. I think this is the best place to update the sufficiency flag, at the same time as inserting the assessment.

File: service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java [hunk 1]
```diff
@@ -105,6 +108,28 @@ public void insertEvidenceSummaryDocument(GeneratePdfPayload request, String doc
     claimRepository.save(claim);
   }
 
+  @Override
+  public void updateSufficientEvidenceFlag(
+      String claimSubmissionId, Boolean flag, String diagnosticCode) {
+    ClaimEntity claim = claimRepository.findByClaimSubmissionId(claimSubmissionId).orElse(null);
+    if (claim == null) {
+      log.warn("Could not find claim with given claimSubmissionId.");
```
Comment (wlin2019) [inline service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java:116]: Is it considered as a problem if the claim is not found? If yes, should it be handled other than just logging here?

File: service/provider/src/main/java/gov/va/vro/service/provider/services/MasAssessmentResultProcessor.java [hunk 2]
```diff
@@ -24,5 +24,9 @@ public void process(Exchange exchange) throws Exception {
       return;
     }
     saveToDbService.insertAssessmentResult(evidence, diagnosticCode);
+    String claimSubmissionId = evidence.getClaimSubmissionId();
+    Boolean sufficiencyFlag = evidence.isSufficientForFastTracking();
+    saveToDbService.updateSufficientEvidenceFlag(
```
Comment (wlin2019) [inline service/provider/src/main/java/gov/va/vro/service/provider/services/MasAssessmentResultProcessor.java:29]: It looks like that the two variables created here don't do much. Just using evidence get methods in next line can reduce a couple of lines.

**Review by wlin2019 — APPROVED** [review]
I'm not familiar with the data model used, and the flow of setting up this flag. I have no objection to merge this PR. But here's some concerns.
1. I feel the DB update logic might be a bit too complicated. But if the data model is set that way, it probably won't be able to find a much better solution.
2. It doesn't seem have proper consideration for error handling.

**Comment by au2** [comment]
@ianwolson is this pr still alive or are the new developments are taken over?

**Pull request closed without merging**